### PR TITLE
Make more effective use of pokemon move macros

### PR DESCRIPTION
**Pull request by Phlosioneer**
I'm open to possible name changes, particularly the prefix. `LEVEL_UP_*` isn't great, but I couldn't think of a better one, so I stuck with the original.

### CODE REVIEW

**Comment by DizzyEggg** [comment]
I'd remove LEVEL_UP_END tbh, 0xFF, 0xFFFF and -1 are everywhere in the repo and they usually mean end of table, so it's kinda self-exploratory.

**Comment by Phlosioneer** [comment]
Normally you're right, but I disagree in this case. The LEVEL_UP_END constant is directly used in if statements and such in not-obvious ways. I think spelling it out makes the code much more clear.

**Comment by Phlosioneer** [comment]
Whoops

**Comment by Diegoisawesome** [comment]
Maybe it would be better to name it something like `INVALID_MOVE`.

**Comment by Phlosioneer** [comment]
This PR has been updated for a while.

File: include/pokemon.h [hunk 1]
```diff
@@ -135,7 +135,7 @@
 #define UNOWN_FORM_COUNT 28
 
 #define LEVEL_UP_MOVE(level, move) ((level << 9) | move)
-#define LEVEL_UP_END 0xFFFF
```
Comment (Diegoisawesome) [inline include/pokemon.h]: This was left in here...

**Pull request closed without merging**